internal/service/cloud/image.go

```diff
@@ -70,10 +71,21 @@ func (s *Service) lookupImageID(ctx context.Context, ms *scope.Machine) (string,
 		images = filterImagesByName(images, version)
 	}
 
-	if n := len(images); n != 1 {
+	if len(images) == 0 {
+		return "", imageMatchError{selector: imageSpec.Selector}
```

Comment:
Since you have an Enum, you can also use a naked switch.
If you want.
not really important.